packages/shared/src/components/buttons/Button.tsx

```diff
@@ -115,7 +133,14 @@ function ButtonComponent<TagName extends AllowedTags>(
     >
       {icon && getIconWithSize(icon)}
       {children && <span className={spanClassName}>{children}</span>}
-      {rightIcon && getIconWithSize(rightIcon)}
+      {rightIcon && rightIcon && (
```

Comment:
This one feels a bit weird, is it really needed?

Reply:
Oh, this was some refactored down code. Forgot to remove the other condition.